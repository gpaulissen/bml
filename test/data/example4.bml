(1N)---
#COPY nt_defense
D   Strength, ca 15+
2C  At least 5-4 majors
  (D)
    P  5+!c, suggestion to play
    R  Asking for better/longer major
    2D 5+!d, suggestion to play
  (P)
    2D Asking for better/longer major
2D  A weak major or a strong minor
  (P)
    2H Pass/correct
    2S Pass/correct
    2N Asking
2HS Constructive
2N  5-5 minors
3X  Preemptive
#ENDCOPY

  Note that the above is only for a direct overcall over 1NT. To
  define the above also when balancing we will use BML's
  copy/cut/paste functionality in order to show that you do not
  have to write it all over again. Take a look below (only visible in
  the /example.txt/ or generated Full Disclosure file, not in HTML or LaTeX):

#HIDE
(1N)-P-(P)---
#PASTE nt_defense
