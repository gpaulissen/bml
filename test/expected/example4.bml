#VUL 00

#SEAT 0

(1N)
  D = Strength, ca 15+
  2C = At least 5-4 majors
    (D)
      P = 5+!c, suggestion to play
      R = Asking for better/longer major
      2D = 5+!d, suggestion to play
    (P)
      2D = Asking for better/longer major
  2D = A weak major or a strong minor
    (P)
      2H = Pass/correct
      2S = Pass/correct
      2N = Asking
  2H = Constructive
  2S = Constructive
  2N = 5-5 minors
  3C = Preemptive
  3D = Preemptive
  3H = Preemptive
  3S = Preemptive

(1N)
  P
    (P)
      D = Strength, ca 15+
      2C = At least 5-4 majors
        (D)
          P = 5+!c, suggestion to play
          R = Asking for better/longer major
          2D = 5+!d, suggestion to play
        (P)
          2D = Asking for better/longer major
      2D = A weak major or a strong minor
        (P)
          2H = Pass/correct
          2S = Pass/correct
          2N = Asking
      2H = Constructive
      2S = Constructive
      2N = 5-5 minors
      3C = Preemptive
      3D = Preemptive
      3H = Preemptive
      3S = Preemptive
