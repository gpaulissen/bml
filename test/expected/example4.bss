*00{}=NYYYYYY
*001N=NYYYYYY0
*001ND=NYYYYYY0Strength, ca 15+
*001N2C=NYYYYYY008At least 5-4 majors
*001N2CD=NYYYYYY0
*001N2CDP=NYYYYYY05+!c, suggestion to play
*001N2CDR=NYYYYYY0Asking for better/longer major
*001N2CD2D=NYYYYYY0085+!d, suggestion to play
*001N2CP=NYYYYYY0
*001N2CP2D=NYYYYYY008Asking for better/longer major
*001N2D=NYYYYYY008A weak major or a strong minor
*001N2DP=NYYYYYY0
*001N2DP2H=NYYYYYY008Pass/correct
*001N2DP2S=NYYYYYY008Pass/correct
*001N2DP2N=NYYYYYY0Asking
*001N2H=NYYYYYY008Constructive
*001N2S=NYYYYYY008Constructive
*001N2N=NYYYYYY05-5 minors
*001N3C=NYYYYYY008Preemptive
*001N3D=NYYYYYY008Preemptive
*001N3H=NYYYYYY008Preemptive
*001N3S=NYYYYYY008Preemptive
*001NP=NYYYYYY0
*001NPP=NYYYYYY0
*001NPPD=NYYYYYY0Strength, ca 15+
*001NPP2C=NYYYYYY008At least 5-4 majors
*001NPP2CD=NYYYYYY0
*001NPP2CDP=NYYYYYY05+!c, suggestion to play
*001NPP2CDR=NYYYYYY0Asking for better/longer major
*001NPP2CD2D=NYYYYYY0085+!d, suggestion to play
*001NPP2CP=NYYYYYY0
*001NPP2CP2D=NYYYYYY008Asking for better/longer major
*001NPP2D=NYYYYYY008A weak major or a strong minor
*001NPP2DP=NYYYYYY0
*001NPP2DP2H=NYYYYYY008Pass/correct
*001NPP2DP2S=NYYYYYY008Pass/correct
*001NPP2DP2N=NYYYYYY0Asking
*001NPP2H=NYYYYYY008Constructive
*001NPP2S=NYYYYYY008Constructive
*001NPP2N=NYYYYYY05-5 minors
*001NPP3C=NYYYYYY008Preemptive
*001NPP3D=NYYYYYY008Preemptive
*001NPP3H=NYYYYYY008Preemptive
*001NPP3S=NYYYYYY008Preemptive
