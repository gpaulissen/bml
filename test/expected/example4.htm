<?xml version="1.0" encoding="utf-8"?>
<!DOCTYPE html PUBLIC '-//W3C//DTD XHTML 1.0 Strict//EN' 'http://www.w3.org/TR/xhtml1/DTD/xhtml1-strict.dtd'>
<html xmlns='http://www.w3.org/1999/xhtml' xml:lang='en' lang='en'><head><meta http-equiv="Content-Type" content="text/html; charset=utf-8" /><title>No title supplied</title><style type="text/css">div.start {
    display: inline-block;
    vertical-align: top;
    width: 60px;
}
    
div.bidtable {
    width:700px;
    padding-bottom: 20px;
}    

div.bidtable ul {
    list-style-type: none;
    margin: 0;
    padding: 0;
}

div.bidtable li ul
{
    list-style-type: none;
    margin: 0;
    padding: 0;
    margin-left: 60px;
}

div.tree {
    list-style-type: none;
    padding-bottom: 20px;
}

div.tree ul {
    list-style-type: none;
    /* vline.png */
    background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAAKAQMAAABPHKYJAAAAA1BMVEWIiIhYZW6zAAAACXBIWXMAAAsTAAALEwEAmpwYAAAAB3RJTUUH1ggGExMZBky19AAAAAtJREFUCNdjYMAEAAAUAAHlhrBKAAAAAElFTkSuQmCC) repeat-y;
    margin: 0;
    padding: 0;
    margin-left: 10px;
}

div.tree li.node {
    margin: 0;
    padding: 0 12px;
    line-height: 20px;
    /* node.png */
    background: url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAUAQMAAACK1e4oAAAABlBMVEUAAwCIiIgd2JB2AAAAAXRSTlMAQObYZgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9YIBhQIJYVaFGwAAAARSURBVAjXY2hgQIf/GTDFGgDSkwqATqpCHAAAAABJRU5ErkJggg==) no-repeat;
}

div.tree td.node {
    vertical-align: top;
    width: 60px;
}

div.tree li.last {
    /* lastnode.png */
    background: #fff url(data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAgAAAAUAQMAAACK1e4oAAAABlBMVEUAAwCIiIgd2JB2AAAAAXRSTlMAQObYZgAAAAlwSFlzAAALEwAACxMBAJqcGAAAAAd0SU1FB9YIBhQIIhs+gc8AAAAQSURBVAjXY2hgQIf/GbAAAKCTBYBUjWvCAAAAAElFTkSuQmCC) no-repeat;
}

div.tree li.root {
    background: #fff none no-repeat;
}

.ccolor{color:Green}
.dcolor{color:Orange}
.hcolor{color:Red}
.scolor{color:Black}
</style></head><body><div class="bidtable"><ul><li><div class="start">(1NT)&ndash;</div><ul><li><div class="start">Dbl</div>Strength, ca 15+</li><li><div class="start">2<span class="ccolor">&clubs;</span></div>At least 5-4 majors<ul><li><div class="start">(D)</div><ul><li><div class="start">Pass</div>5+<span class="ccolor">&clubs;</span>, suggestion to play</li><li><div class="start">Rdbl</div>Asking for better/longer major</li><li><div class="start">2<span class="dcolor">&diams;</span></div>5+<span class="dcolor">&diams;</span>, suggestion to play</li></ul></li><li><div class="start">(P)</div><ul><li><div class="start">2<span class="dcolor">&diams;</span></div>Asking for better/longer major</li></ul></li></ul></li><li><div class="start">2<span class="dcolor">&diams;</span></div>A weak major or a strong minor<ul><li><div class="start">(P)</div><ul><li><div class="start">2<span class="hcolor">&hearts;</span></div>Pass/correct</li><li><div class="start">2<span class="scolor">&spades;</span></div>Pass/correct</li><li><div class="start">2NT</div>Asking</li></ul></li></ul></li><li><div class="start">2<span class="hcolor">&hearts;</span><span class="scolor">&spades;</span></div>Constructive</li><li><div class="start">2NT</div>5-5 minors</li><li><div class="start">3X</div>Preemptive</li></ul></li></ul></div></body></html>